Mathlib/Logic/Equiv/Fin.lean

```diff
@@ -498,3 +498,8 @@ instance subsingleton_fin_zero : Subsingleton (Fin 0) :=
 /-- `Fin 1` is a subsingleton. -/
 instance subsingleton_fin_one : Subsingleton (Fin 1) :=
   finOneEquiv.subsingleton
+
+/-- The natural `Equiv` between `(Fin m → X) × (Fin n → X)` and `(Fin (m + n) → X)`.-/
+def Equiv.finArrowProdEquivFinAddArrow {α : Type*} (m n : ℕ) :
+    (Fin m → α) × (Fin n → α) ≃ (Fin (m + n) → α) :=
+  (sumArrowEquivProdArrow _ _ _).symm.trans (finSumFinEquiv.arrowCongr (Equiv.refl _))
```

Comment:
I think it might be best to construct this explicitly with `Fin.append`, rather than build it through composition; that way you get the fact that it is equal to `Fin.append` by definition.